include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -2352,6 +2359,24 @@ ::fill_JxW_values(AlignedVector<VectorizedArray<Number> > &JxW_values) const
 
 
 
+template <int dim, int n_components_, typename Number>
+inline
+VectorizedArray<Number>
+FEEvaluationBase<dim,n_components_,Number>::JxW(const unsigned int q_point) const
+{
+  Assert (this->J_value != 0, ExcNotImplemented());
+  if (this->cell_type == internal::MatrixFreeFunctions::cartesian ||
+      this->cell_type == internal::MatrixFreeFunctions::affine)
+    {
+      Assert (this->mapping_info != 0, ExcNotImplemented());
```

Comment:
This one should actually have been an internal error. You are right about the other ones.